### PR TITLE
ENH: errorbar color cycle clean up

### PR DESCRIPTION
**Pull request by tacaswell**
@mdboom This is what I was talking about on the phone call.

### CODE REVIEW

**Comment by mdboom** [comment]
Great -- this definitely addresses part of #5584 in a better way.  Should we also remove the use of `Axes.plot` in favor of `Axes.line2d` while we're in there?

**Comment by tacaswell** [comment]
The decision to consult the prop-cycle still needs to be fixed.

attn @WeatherGod

**Comment by tacaswell** [comment]
and is this a 2.0 or 2.1 tagged issue?

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -2833,45 +2852,68 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
+                eb_cap_style[key] = kwargs[key]
+        eb_cap_style['color'] = ecolor
 
-        if not iterable(uplims):
-            uplims = np.array([uplims] * len(x), bool)
-        else:
-            uplims = np.asarray(uplims, bool)
+        if plot_line:
+            l0 = mlines.Line2D(x, y, **plot_line_style)
+            self.add_line(l0)
```
Comment (jenshnielsen) [inline lib/matplotlib/axes/_axes.py]: Nitpick this variable name looks a bit to close to `10` I first read the line as `self.add_line(10)`

Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: Indeed, will fix that.

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -2833,45 +2852,68 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
+            eb_cap_style['markeredgewidth'] = capthick
 
-        # arrays fine here, they are booleans and hence not units
-        if not iterable(lolims):
-            lolims = np.asarray([lolims] * len(x), bool)
-        else:
-            lolims = np.asarray(lolims, bool)
+        # For backwards-compat, allow explicit setting of
+        # 'markeredgewidth' to over-ride capthick.
+        for key in ('markeredgewidth', 'transform', 'alpha',
```
Comment (jenshnielsen) [inline lib/matplotlib/axes/_axes.py]: Are we normalising `mew` out somewhere else?

**Comment by jenshnielsen** [comment]
👍 @tacaswell Does this need more work ([WIP] in the title) of is it ready?

**Comment by tacaswell** [comment]
This should be ready to go.

**Comment by jenshnielsen** [comment]
Back port to 2.x as acfab37c923c04c3fb9e2d1d9970fecc0b6a14f1

**Pull request merged**